domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -73,10 +222,114 @@ class ProfileManagementController @Inject constructor(
+    val pendingLiveData = MutableLiveData(AsyncResult.pending<Any?>())
+    val deferred = profileDataStore.storeDataWithCustomChannelAsync(updateInMemoryCache = true) {
+      val profile = it.profilesMap[profileId.internalId] ?: return@storeDataWithCustomChannelAsync Pair(it, ProfileActionStatus.PROFILE_NOT_FOUND)
+      val updatedProfile = profile.toBuilder().setAllowDownloadAccess(allowDownloadAccess).build()
+      val profileDatabaseBuilder = it.toBuilder().putProfiles(profileId.internalId, updatedProfile)
+      Pair(profileDatabaseBuilder.build(), ProfileActionStatus.SUCCESS)
+    }
+    deferred.invokeOnCompletion {
+      if (it != null) {
+        logger.e("ProfileManagementController", "Failed to update allowDownloadAccess", it)
+        pendingLiveData.postValue(AsyncResult.failed(it))
+      } else {
+        when (deferred.getCompleted()) {
+          ProfileActionStatus.SUCCESS -> pendingLiveData.postValue(AsyncResult.success(null))
+          else -> pendingLiveData.postValue(AsyncResult.failed(ProfileNotFoundException("ProfileId ${profileId.internalId} does not match an existing Profile")))
+        }
+      }
+    }
+    return pendingLiveData
+  }
+
+  /**
+   * Login to the user's Profile by setting the current profile Id and updating profile's last logged in time.
+   *
+   * @param profileId the ID corresponding to the profile being logged into.
+   * @return a [LiveData] that indicates the success/failure of this login operation.
+   */
+  @ExperimentalCoroutinesApi
+  fun loginToProfile (profileId: ProfileId): LiveData<AsyncResult<Any?>> {
+    val pendingLiveData = MutableLiveData(AsyncResult.pending<Any?>())
+    setCurrentProfileId(profileId).observeForever { setIdResult ->
+      if (setIdResult.isSuccess()) {
+        updateLastLoggedIn(profileId).observeForever { updateLoggedInResult ->
+          if (updateLoggedInResult.isSuccess()) {
+            pendingLiveData.postValue(AsyncResult.success(null))
+          } else if (updateLoggedInResult.isFailure()) {
+            pendingLiveData.postValue(AsyncResult.failed(updateLoggedInResult.getErrorOrNull()!!))
+          }
+        }
+      } else if (setIdResult.isFailure()) {
+        pendingLiveData.postValue(AsyncResult.failed(setIdResult.getErrorOrNull()!!))
+      }
+    }
+    return pendingLiveData
+  }
+
+  /**
+   * Sets the currentProfileId to the selected profile in ProfileChooserFragment.
+   * Checks to ensure that profileId corresponds to an existing profile.
+   *
+   * @param profileId the ID corresponding to the profile being set.
+   * @return a [LiveData] that indicates the success/failure of this set operation.
+   */
+  internal fun setCurrentProfileId(profileId: ProfileId): LiveData<AsyncResult<Any?>> {
+    val pendingLiveData = MutableLiveData(AsyncResult.pending<Any?>())
+    val profileLiveData = dataProviders.convertToLiveData(profileDataStore)
+    // TODO: Update DataProviders to allow for a one time access read operation
```

Comment:
Please resolve in this PR or associate an issue with it.

Reply:
Removed comment because setCurrentProfileId no longer uses live data.